thunder/__init__.py

```diff
@@ -595,11 +595,24 @@ def get_computation_and_inputs(*args, **kwargs):
                     computation_traces.append(computation_trc)
                 cs.last_computation_transformation_stop = time.perf_counter_ns()
 
+                from thunder.executors.passes import transform_for_execution as transform_for_execution_pass
+                from thunder.executors.passes import _transform_for_operator_executor_execution
+                from thunder.distributed.utils import maybe_sort_waits
+
+                with langctxs.langctx(cd.langctx):
+                    tmp_comp_trc = _transform_for_operator_executor_execution(computation_trc, cd.executors_list)
```

Comment:
we have this `_transform_for_operator_executor_execution` also in the `transform_for_execution`, does it affect anything? I noticed the bucketing was applied before `_transform_for_operator_executor_execution`, now it's after the `_transform_for_operator_executor_execution`, don't know if it matters

Reply:
To effectively apply `sort_waits` to a trace, `wait`s need to be a BoundSymbol. If a trace has a BoundSymbol representing an in-place distributed op, its subsymbols are a pair of out-of-place dist op, wait, and copy and there does not exist a boundsymbol whose sym is wait.
The call of `_transform_for_operator_executor_execution` here flattens such bsyms of in-place dist ops and `tmp_comp_trc` would have bsyms of waits, if the computation_trc has bsyms of in-place dist ops.